hazelcast/src/main/java/com/hazelcast/map/impl/LockAwareLazyMapEntry.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright (c) 2008-2017, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.map.impl;
+
+import com.hazelcast.internal.serialization.InternalSerializationService;
+import com.hazelcast.nio.serialization.Data;
+import com.hazelcast.query.impl.getters.Extractors;
+
+/**
+ * Specialization of the LazyMapEntry that is LockAware. Used in EntryProcessor.
+ * If serialized will be deserialized as the LazyMapEntry, since the locked information is volatile, valid to the run only.
+ */
+public class LockAwareLazyMapEntry extends LazyMapEntry implements LockAware {
```

Comment:
The test failure was because `LockAwareLazyMapEntry` extends `LazyMapEntry` which is IDS :
- it has no default constructor
- it has the same type ID